WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java

```diff
@@ -282,17 +282,17 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         switch (requestCode) {
             case RequestCodes.PICTURE_LIBRARY:
             case RequestCodes.VIDEO_LIBRARY:
-                Uri imageUri = data.getData();
-                String mimeType = getContentResolver().getType(imageUri);
-                fetchMedia(imageUri, mimeType);
+                if (resultCode == Activity.RESULT_OK && data != null) {
+                    Uri imageUri = data.getData();
+                    String mimeType = getContentResolver().getType(imageUri);
+                    fetchMedia(imageUri, mimeType);
+                }
                 break;
             case RequestCodes.TAKE_PHOTO:
-                if (resultCode == Activity.RESULT_OK) {
-                    if (data != null) {
-                        Uri uri = Uri.parse(mMediaCapturePath);
-                        mMediaCapturePath = null;
-                        queueFileForUpload(uri, getContentResolver().getType(uri));
-                    }
+                if (resultCode == Activity.RESULT_OK && data != null) {
```

Comment:
I didn't notice this before, but is the check for `data != null` necessary here since `data` is unused?